### ISSUE TITLE
Problem with initialization checker and generic parent class

### ISSUE THREAD

**Issue by twistedsquare** [typetools/checker-framework#2052]
I have an issue with the initialization checker and generic parent classes.  Given this parent class:

```
public class Parent<S>
{
    protected final String field = "";
}
```

And this child class:

```
import org.checkerframework.checker.initialization.qual.UnknownInitialization;

public class Child extends Parent<String>
{
    public String getField(@UnknownInitialization(Parent.class) Child this)
    {
        return this.field;
    }
}
```

I get an error that the return of getField can be null, even though field is initialised by the Parent class, which is declared to be initialized at that point.  This seems like a false negative to me: an error on safe code.  (Tested on 2.5.2)

This seems to relate to the parent being generic -- if you remove the type parameter from Parent (and the corresponding parameter from Child's extends clause), the error goes away.  So I think it's perhaps that the `Parent.class` literal is not being matched to the `Parent<String>` actual parent?  In theory, I think `Parent.class` should unambigiously refer to `Parent<String>` even if it's using the raw type, because I think each erased class can only occur once in a given class's inheritance hierarchy?

One additional note.  In trying to minimise this example, I put both classes as static inner classes in a parent class (which I think I've seen done in checker framework tests to make the issue self-contained).  But if I do that, the error goes away:

```
import org.checkerframework.checker.initialization.qual.UnknownInitialization;

public class WrappedExample
{
    public static class ParentW<S>
    {
        protected final String field = "";
    }
    
    public static class ChildW extends ParentW<String>
    {
        public String getField(@UnknownInitialization(ParentW.class) ChildW this)
        {
            return this.field;
        }
    }
}
```

It feels a bit strange that the inner-class version gives a different checker outcome to the separate-class version.  It seems it's to do with initialising the field directly inline; if I move the initialisation of field into a new ParentW constructor I get the same nullness error as the separate-class version.  Maybe that's just a side effect of how Java compilation proceeds, but it seems relevant if the checker framework usually puts its bugs into inner classes.
